koku/api/report/test/aws/tests_queries.py

```diff
@@ -1268,3 +1268,15 @@ def test_execute_query_with_tag_filter(self):
         for key in totals:
             result = data_totals.get(key, {}).get('value')
             self.assertEqual(result, totals[key])
+
+    def test_prefixed_logical_and_or(self):
+        """"Test group_by[and:account] prefixes."""
+        """Test execute_query for current month on daily breakdown."""
+        url = '?group_by[and:account]=7777&group_by[and:account]=4444'
```

Comment:
Instead of hardcoding two account numbers you should use ones from the unit test. In the setup one is created for you: `self.fake_aws.account_alias`

In the current case you are likely to only have 0 cost data, since those account probably don't exist for this tenant.

You can create additional accounts with the following steps for this test case:
```
redhatisgreat_fake_aws = FakeAWSCostData(self.provider, account_alias='redhatisgreat')
redhatisgreat_generator = AWSReportDataGenerator(self.tenant)
redhatisgreat_generator.add_data_to_tenant(redhatisgreat_fake_aws)

redhatisthefuture_fake_aws = FakeAWSCostData(self.provider, account_alias='redhatisthefuture')
redhatisthefuture_generator = AWSReportDataGenerator(self.tenant)
redhatisthefuture_generator.add_data_to_tenant(redhatisthefuture_fake_aws)
```
So you can reference the account_alias values from in this URL.
url = '?group_by[and:account]=redhat&group_by[and:account]=future'

So its important that the test case prove that the *and* condition is working.

Here you can run the individual group_by query for the `redhatisthefuture` account and `redhatisgreate` account on its own and collect that total cost.
`url = '?group_by[account]=redhatisthefuture'`
`url = '?group_by[account]=redhatisgreat'`

You can run a the default OR case
`url = '?group_by[account]=redhat&group_by[account]=future'`

Then you can run the and condition above:
url = '?group_by[and:account]=redhat&group_by[and:account]=future'
which should only match the total for `redhatisthefuture`.

Reply:
added